src/kboot_gpu.c

```diff
@@ -37,6 +38,18 @@ struct aux_perf_states {
     struct aux_perf_state states[];
 };
 
+int32_t rust_gpu_initdata_size(uint32_t compat_maj, uint32_t compat_min, size_t *data_a,
+                               size_t *data_b, size_t *globals);
+
+int32_t rust_fill_gpu_initdata(size_t perf_state_table_count, size_t perf_state_count,
```

Comment:
can we use a struct for this?

Reply:
We can, i just find structs to be a bit inconvenient to keep in sync with rust/c

Reply:
but keeping 22 function arguments in sync is convenient? I find the the function argument variant harder to validate and would assume it's more likely to attract bugs. We could use bindgen.

Reply:
no, they are both terrible tbh. i'll look into bindgen